### PR TITLE
[KIWI-2035] - Update name assignment logic

### PR DESCRIPTION
**Pull request by Stelios-CB**
### What changed

Updated name assignment logic to now account for First, Middle and Last names.

### Why did it change

Experian /verify doesn't use middle names, it was necessary to extract the first name from the values provided in `shared_claims`.

### Issue tracking

- [KIWI-2035](https://govukverify.atlassian.net/browse/KIWI-2035)

[KIWI-2035]: https://govukverify.atlassian.net/browse/KIWI-2035?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by GDainton — DISMISSED** [review]

File: src/services/VerifyAccountRequestProcessor.ts [hunk 1]
```diff
@@ -147,7 +148,7 @@ export class VerifyAccountRequestProcessor {
   		restricted: {
   			Experian_request_details: [
```
Comment (gds-jbentham) [inline src/services/VerifyAccountRequestProcessor.ts:149]: Is this up to date? It doesnt match whats in the request catalogue

File: src/tests/unit/utils/PersonIdentityUtils.test.ts [hunk 2]
```diff
@@ -1,33 +1,43 @@
...
+		const result = getMiddleNames([{ nameParts: [
+			{ type: "GivenName", value: "First" },
+			{ type: "GivenName", value: "MiddleOne" },
+			{ type: "GivenName", value: "MiddleTwo" },
+			{ type: "FamilyName", value: "Last" },
+		] }]);
+		expect(result).toBe("MiddleOne MiddleTwo");
 	});
+
+	it("returns last name", () => {
```
Comment (gds-jbentham) [inline src/tests/unit/utils/PersonIdentityUtils.test.ts:33]: Can we add a test for no first name and empty name to make the sure the code performs correctly

Comment (gds-jbentham) [reply src/tests/unit/utils/PersonIdentityUtils.test.ts:33]: Also what happens if we call getMiddleNames with just one middle name

File: src/services/VerifyAccountRequestProcessor.ts (hunk 1 above)
Comment (Stelios-CB) [reply src/services/VerifyAccountRequestProcessor.ts]: Updated

File: src/tests/unit/utils/PersonIdentityUtils.test.ts (hunk 2 above)
Comment (Stelios-CB) [reply src/tests/unit/utils/PersonIdentityUtils.test.ts:52]: First name / family names are required fields in CIC, is it realistic that we would ever receive these as empty values?

`getMiddleNames` will respond with a string containing single or multiple values depending on the payload, and an empty string if none are provided

File: src/services/ExperianService.ts [hunk 3]
```diff
@@ -82,7 +82,8 @@ export class ExperianService {
     							},
     							names: [
     								{
-    									firstName: givenName,
+    									firstName: firstName,
+										middleNames: "",
```
Comment (gds-jbentham) [inline src/services/ExperianService.ts:86]: Rather than pass an empty string can we just exclude this field.

File: src/services/VerifyAccountRequestProcessor.ts [hunk 4]
```diff
@@ -143,20 +143,20 @@ export class VerifyAccountRequestProcessor {
...
   		restricted: {
-  			Experian_request_details: [
-  				{
-  					name:`${givenName} ${surname}`,
-  					sortCode: verifyAccountPayload?.sort_code,
-  					accountNumber: verifyAccountPayload?.account_number,
-  					attemptNum: session.attemptCount ?? 1,
-  				},
-  			],
+  			name: person.name,
```
Comment (gds-jbentham) [inline src/services/VerifyAccountRequestProcessor.ts:151]: Will this not also return the middlenames?

Comment (Stelios-CB) [reply src/services/VerifyAccountRequestProcessor.ts:151]: Yep, it returns the entire name object we receive in `shared_claims` which includes all 'GivenNames', this is in line with the [event catalogue](https://event-catalogue.internal.account.gov.uk/events/BAV_EXPERIAN_REQUEST_SENT/)

File: src/services/ExperianService.ts (hunk 3 above)
Comment (Stelios-CB) [reply src/services/ExperianService.ts]: Done

File: src/services/VerifyAccountRequestProcessor.ts (hunk 4 above)
Comment (gds-jbentham) [reply src/services/VerifyAccountRequestProcessor.ts:151]: This needs to reflect what we send to Experian so should exclude the middle names

**Review by gds-jbentham — DISMISSED** [review]

**Review by gds-jbentham — APPROVED** [review]
